label/label_blackbox_test.go

```diff
@@ -103,6 +105,59 @@ func (s *TestLabelRepository) TestCreateLabelWithWrongColorCode() {
 	assert.Contains(s.T(), err.Error(), "labels_background_color_check")
 }
 
+func (s *TestLabelRepository) TestSaveLabel() {
+	testFxt := tf.NewTestFixture(s.T(), s.DB, tf.Labels(1))
+	repo := label.NewLabelRepository(s.DB)
+	l := testFxt.Labels[0]
+	l.Name = "severity/p5"
+	l.TextColor = "#778899"
+	l.BackgroundColor = "#445566"
+	l.BorderColor = "#112233"
+
+	lbl, err := repo.Save(context.Background(), *l)
+	require.Nil(s.T(), err)
+	assert.Equal(s.T(), l.Name, lbl.Name)
+	assert.Equal(s.T(), l.TextColor, lbl.TextColor)
+	assert.Equal(s.T(), l.BackgroundColor, lbl.BackgroundColor)
+	assert.Equal(s.T(), l.BorderColor, lbl.BorderColor)
+}
+
+func (s *TestLabelRepository) TestSaveLabelNonExisting() {
+	fakeID := uuid.NewV4()
+	fakeLabel := label.Label{
+		ID: fakeID,
+	}
+	repo := label.NewLabelRepository(s.DB)
+	_, err := repo.Save(context.Background(), fakeLabel)
+	require.NotNil(s.T(), err)
+	assert.Equal(s.T(), reflect.TypeOf(errs.NotFoundError{}), reflect.TypeOf(err))
+}
+
+func (s *TestLabelRepository) TestUpdateLabelWithSameName() {
+	testFxt := tf.NewTestFixture(s.T(), s.DB, tf.Labels(1))
+	repo := label.NewLabelRepository(s.DB)
+	name := "TestUpdateLabel"
+	l := label.Label{
+		SpaceID: testFxt.Spaces[0].ID,
+		Name:    name,
+	}
+	repo.Create(context.Background(), &l)
+	require.NotEqual(s.T(), uuid.Nil, l.ID)
+	require.Equal(s.T(), "#000000", l.TextColor)
```

Comment:
You MUST NOT rely on values that you didn't explicitly set when creating the label with a test fixture. That might change its defaults some time.

Reply:
Fixed. 9945b9941f1241159878fc83faeb0a83ca037f9b